### PR TITLE
Documentation improvements

### PR DESCRIPTION
**Pull request by matthchr**

### CODE REVIEW

File: docs/v2/adding-new-codegenerated-resource.md [hunk 1]
```diff
@@ -63,7 +63,31 @@ See [the code generator README](/hack/README.md) for how to run the code generat
 TODO: expand on common errors
 
 ## Examine the generated resource
-The Azure REST API specs and ARM template JSON schemas are not perfect. Sometimes they mark a `readonly` property as mutable or have another error or mistake. Have a look through the [generated code](/hack/generated/apis/) in the directory named after the `group` of the resource that was added. In our `NetworkSecurityGroups` example, this is at `/hack/generated/apis/microsoft.network/v1alpha1api20201101/network_security_group_types_gen.go`
+After running the generator, the new resource you added should be in the [apis](/hack/generated/apis/) directory. 
+
+Have a look through the  in the directory named after the `group` of the resource that was added.
```
Comment (theunrepentantgeek) [inline docs/v2/adding-new-codegenerated-resource.md:68]: Missing words

```suggestion
Have a look through the files in the directory named after the `group` and `version` of the resource that was added.
```

File: hack/CONTRIBUTING.md [hunk 2]
```diff
@@ -92,3 +93,10 @@ Run the following commands to create a `kind` cluster and deploy the operator, C
 4. `task controller:install` - Installs the CRDs, webhooks, and operator pod (running the image from the local registry) into the cluster.
 
 Once these steps have completed successfully you can use `kubectl` to interact with the local `kind` cluster.
+
+## Common problems and their solutions
+
+### error loading schema from root... open /azure-service-operator/hack/generator/specs/azure-resource-manager-schemas/schemas/2019-04-01/deploymentTemplate.json no such file or directory
```
Comment (theunrepentantgeek) [inline hack/CONTRIBUTING.md:99]: Seems a bit long for a heading

```suggestion
### Error loading schema from root

Full error text:
error loading schema from root... open /azure-service-operator/hack/generator/specs/azure-resource-manager-schemas/schemas/2019-04-01/deploymentTemplate.json no such file or directory
```

**Review by theunrepentantgeek — COMMENTED** [review]
Looks good.

gif

File: hack/CONTRIBUTING.md (hunk 2 above)
Comment (matthchr) [reply hack/CONTRIBUTING.md:99]: Good idea

File: docs/v2/adding-new-codegenerated-resource.md (hunk 1 above)
Comment (matthchr) [reply docs/v2/adding-new-codegenerated-resource.md:68]: Good eye

**Review by theunrepentantgeek — APPROVED** [review]

**Pull request merged**